assert/utils.go

```diff
@@ -947,6 +947,19 @@ func formatNumericComparisonError(actual, expected interface{}, operation string
 	return msg.String()
 }
 
+func formatBeWithinError[T Float](actual T, expected T, tolerance T) string {
+	diff := math.Abs(float64(actual) - float64(expected))
+
+	var msg strings.Builder
+	msg.WriteString(fmt.Sprintf("Expected value to be close to %.4f (±%.4f), but it was not:\n", expected, tolerance))
```

Comment:
Also when someone provides 1.23456789 being close to 1.234569999 with a 0.0000001

You are likely to display that 1.2345 should be close to 1.2345 (+/- whatever)

It's right but not actionable and a bit confusing.

Here you should consider displaying how the difference is lower than the threshold

It should be tested